Modules/_io/_iomodule.c

```diff
@@ -388,7 +388,7 @@ _io_open_impl(PyObject *module, PyObject *file, const char *mode,
         goto error;
 
     /* buffering */
-    {
+    if (buffering < 0) {
```

Comment:
I would prefer to have an else block to initialize isatty variable, even if it's not used in practice. ("else { isatty = 0; }"). Or initialize the variable to 0 when it's declared?

It should prevent warning from dummy compilers.

Reply:
Or add
```c
if (isatty) {
    buffering = 1;
}
```
and simplify the condition `(buffering == 1 || (buffering < 0 && isatty))` below to just `(buffering == 1)`.

Reply:
Changed. `isatty` is now initialized and the redundant `buffering < 0` below has been removed.